src/Simplify.elm

```diff
@@ -1352,10 +1353,64 @@ expressionVisitorHelp node context =
                 _ ->
                     onlyErrors []
 
+        Expression.RecordAccess record field ->
+            let
+                fieldName =
+                    Node.value field
+            in
+            case Node.value record of
+                Expression.RecordExpr setters ->
+                    onlyErrors (recordAccessChecks node fieldName setters)
+
+                Expression.RecordUpdateExpression _ setters ->
+                    onlyErrors (recordAccessChecks node fieldName setters)
```

Comment:
You don't need the entire node here, only it's range
```suggestion
                    onlyErrors (recordAccessChecks (Node.range node) fieldName setters)

                Expression.RecordUpdateExpression _ setters ->
                    onlyErrors (recordAccessChecks (Node.range node) fieldName setters)
```

Reply:
Fair enough!